python/lsst/cp/verify/repackStats.py

```diff
@@ -0,0 +1,448 @@
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import numpy as np
+
+from astropy.table import Table
+
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+import lsst.pex.config as pexConfig
+
+__all__ = [
+    "CpVerifyRepackConnections",
+    "CpVerifyRepackConfig",
+    "CpVerifyRepackBiasTask",
+    "CpVerifyRepackDarkTask",
+    "CpVerifyRepackFlatTask",
+    "CpVerifyRepackDefectTask",
+    "CpVerifyRepackPtcTask",
+    "CpVerifyRepackBfkTask",
+    "CpVerifyRepackCtiTask",
+]
+
+
+class CpVerifyRepackConnections(pipeBase.PipelineTaskConnections,
+                                dimensions={"instrument"},
+                                defaultTemplate={}):
+    detectorStats = cT.Input(
+        name="detectorStats",
+        doc="Input detector statistics.",
+        storageClass="StructuredDataDict",
+        dimensions={"instrument", "exposure", "detector"},
+        multiple=True,
+    )
+    exposureStats = cT.Input(
+        name="exposureStats",
+        doc="Input exposure statistics.",
+        storageClass="StructuredDataDict",
+        dimensions={"instrument", "exposure"},
+        multiple=True,
+    )
+    runStats = cT.Input(
+        name="runStats",
+        doc="Input Run statistics.",
+        storageClass="StructuredDataDict",
+        dimensions={"instrument"},
+        multiple=True,
+    )
+
+    outputCatalog = cT.Output(
+        name="cpvCatalog",
+        doc="Output merged catalog.",
+        storageClass="ArrowAstropy",
+        dimensions={"instrument"},
+    )
+
+
+class CpVerifyRepackConfig(pipeBase.PipelineTaskConfig,
+                           pipelineConnections=CpVerifyRepackConnections):
+
+    expectedDistributionLevels = pexConfig.ListField(
+        dtype=float,
+        doc="Percentile levels expected in the calibration header.",
+        default=[0, 5, 16, 50, 84, 95, 100],
+    )
+
+
+class CpVerifyRepackTask(pipeBase.PipelineTask):
+    """Repack cpVerify statistics for analysis_tools.
+    """
+    ConfigClass = CpVerifyRepackConfig
+    _DefaultName = "cpVerifyRepack"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+
+        inputs["detectorDims"] = [exp.dataId.byName() for exp in inputRefs.detectorStats]
```

Comment:
This should be `dict(exp.dataId.required)` instead of `exp.dataId.byName()`

Reply:
That explains the warnings I'm seeing.  I'm going to wait for the next ticket to fix this, as I have been struggling to get this ticket to pass Jenkins for the merge.